src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V2/DownloadedPackagesStoreV2.swift

```diff
@@ -65,31 +65,50 @@ protocol DownloadedPackagesStoreV2: AnyObject {
 	/// - Parameter etags: The list of Etags to check for
 	func validateCachedKeyPackages(revokationList etags: [String]) throws
 
-
-	#if !RELEASE
 	var keyValueStore: Store? { get set }
-	#endif
 }
 
 extension DownloadedPackagesStoreV2 {
 
 	func addFetchedDays(_ dayPackages: [String: PackageDownloadResponse], country: Country.ID) throws {
+		var revokedPackageError: Error?
 		try dayPackages.forEach { day, bucket in
-			try self.set(country: country, day: day, etag: bucket.etag, package: bucket.package)
+			do {
+				try self.set(country: country, day: day, etag: bucket.etag, package: bucket.package)
+			} catch DownloadedPackagesSQLLiteStore.StoreError.revokedPackage {
+				revokedPackageError = DownloadedPackagesSQLLiteStore.StoreError.revokedPackage
+			}
+		}
+		if let revokedPackageError = revokedPackageError {
+			throw revokedPackageError
```

Comment:
I think we might make this a little bit more compact by:
```
	func addFetchedDays(_ dayPackages: [String: PackageDownloadResponse], country: Country.ID) throws {
		try dayPackages.forEach { day, bucket in
			do {
				try self.set(country: country, day: day, etag: bucket.etag, package: bucket.package)
			} catch DownloadedPackagesSQLLiteStore.StoreError.revokedPackage {
				throw DownloadedPackagesSQLLiteStore.StoreError.revokedPackage
			}
		}
	}
```

Reply:
It's not the same result with you suggested code.
Think of a package in the middle of the list which gets revoked.
What we want, is that the rest of the list is still persisted.
This is why "collect" the revokedPackageError and throw it later.